### PR TITLE
[CAM-8436]-Update-Instance-APIDoc

### PR DESCRIPTION
**Pull request by jtorresjj**
@gsanchezu @julio-castanar 
Review this pull request.

### CODE REVIEW

File: Application Lifecycle Management/instances-api.md [hunk 1]
```diff
@@ -6,65 +6,81 @@
 "contentIsHTML": false
```
Comment (julio-castanar) [inline Application Lifecycle Management/instances-api.md]: Add your name and update date (MM-DD-YY)

File: Application Lifecycle Management/instances-api.md [hunk 2]
```diff
@@ -6,65 +6,81 @@
...
+| [GET /services/instances/{instance_id}/activity](#get-servicesinstancesinstance_idactivity) | Gets all activity logs from the executed operations of an instance. |
+| [GET /services/instances/{instance_id}/machine_logs](#get-servicesinstancesinstance_idmachine_logs) | Gets the logs of all machines of a deployed instance. |
+| [GET /services/instances/{instance_id}/bindings](#get-servicesinstancesinstance_idbindings) | Gets the binding of a instance. |
+| [GET /services/instances/{instance_id}/operations](#get-servicesinstancesinstance_idoperations) | Gets all operations of an instance. |
+| [PUT /services/instances/{instance_id}/deploy](#put-servicesinstancesinstance_iddeploy) | Re-deploy an existing instance. |
+| [PUT /services/instances/{instance_id}/poweron](#put-servicesinstancesinstance_idpoweron) | Power-on an existing instance. |
+| [PUT /services/instances/{instance_id}/shutdown](#put-servicesinstancesinstance_idshutdown) | 	Shutdown an existing instance. |
+| [PUT /services/instances/{instance_id}/   ](#put-servicesinstancesinstance_idreinstall) | Re-install an existing instance. |
```
Comment (julio-castanar) [inline Application Lifecycle Management/instances-api.md]: PUT /services/instances/{instance_id}/reinstall

File: Application Lifecycle Management/instances-api.md [hunk 3]
```diff
@@ -307,21 +326,59 @@ ElasticBox-Release: 4.0
...
+#### Structure
+```
+[POST] /services/instances
+```
 
-**Error Response Codes**
+#### Example
+```
+[POST] https://cam.ctl.io/services/instances
+```
```
Comment (julio-castanar) [inline Application Lifecycle Management/instances-api.md]: Place this block (lines 344-370) after Request - Uri Parameters table

File: Application Lifecycle Management/instances-api.md [hunk 4]
```diff
@@ -333,85 +390,170 @@ Creates a new instance and gets the created instance.
...
-       }]
-   },
-   "instance_tags": [
 
-   ],
-   "automatic_updates": "off",
-   "policy_box": {
-       "id": "e57466ee-7094-4bd4-9121-a6df4395d493",
-       "variables": [
+#### Response Body
```
Comment (julio-castanar) [inline Application Lifecycle Management/instances-api.md]: Add here next lines with section Response Parameters
#### Response Parameters
See description in previous command

File: Application Lifecycle Management/instances-api.md [hunk 5]
```diff
@@ -333,85 +390,170 @@ Creates a new instance and gets the created instance.
...
-           "visibility": "public"
-       }]
-   },
-   "instance_tags": [
 
-   ],
-   "automatic_updates": "off",
-   "policy_box": {
-       "id": "e57466ee-7094-4bd4-9121-a6df4395d493",
-       "variables": [
```
Comment (julio-castanar) [inline Application Lifecycle Management/instances-api.md]: Add here next lines with section Response Parameters

###Response Parameters
See description in previous command

File: Application Lifecycle Management/instances-api.md [hunk 6]
```diff
@@ -333,85 +390,170 @@ Creates a new instance and gets the created instance.
...
+[POST] /services/instances
+```
 
-Content-Type: application/json
-Authorization: Bearer your_json_web_token
-ElasticBox-Release: 4.0
+#### Example
+```
+[POST] https://cam.ctl.io/services/instances
 ```
```
Comment (julio-castanar) [inline Application Lifecycle Management/instances-api.md]: Place this block (lines 556-606) after Request - Uri Parameters table of this command

File: Application Lifecycle Management/instances-api.md [hunk 7]
```diff
@@ -747,165 +943,183 @@ ElasticBox-Release: 4.0
...
+Authorization: Bearer your_json_web_token
+ElasticBox-Release: 4.0
+```
 
-**Request Parameters**
+#### URI Parameters
 
-| Parameter | Type |Description |
-|-----------|------|------------|
+|Parameter |Type |Description | Req. |
```
Comment (julio-castanar) [inline Application Lifecycle Management/instances-api.md]: Add Required column values in this table

File: Application Lifecycle Management/instances-api.md [hunk 8]
```diff
@@ -1120,68 +1402,6 @@ In this sample request, the instance is tagged and scheduled to terminate at a g
...
-| state | string | Instance state, there are three possible states: processing , done and unavailable |
-| members | array | Instance members. |
-| owner | string | Instance owner. |
-| operation | string | Last operation, there are seven types of operations: deploy , shutdown , poweron , reinstall , reconfigure , terminate and terminate_service |
-| icon | string | Instance icon uri. |
-| id | array | Instance unique identifier. |
-| schema | string | Instance schema uri. |
-| lease | array | Schedules an instance with two parameters:<li>released. Is a true or false boolean value. False means that the operation scheduled on the instance is not executed yet.</li><li>expire. Specifies in UTC format YYYY-MM-DD HH:MM:SS.SSSSSS, the time and date for stopping an instance. It’s required only when an instance is set to terminate or shut down.</li><li>operation. Specifies an instance to stop with **shutdown** or **terminate**. When not scheduled, the instance is set to **alwayson**.</li> |
 
 ```
```
Comment (julio-castanar) [inline Application Lifecycle Management/instances-api.md]: Did you paste this block twice? (lines 1406-1535)
Remove it or comment it before, if it's a new example

File: Application Lifecycle Management/instances-api.md [hunk 9]
```diff
@@ -1497,89 +1790,304 @@ GET /services/instances/{instance_id}/activity?operation=deploy
 | created | string | Activity creation date. |
 |* event | string | Activity event. |
 
+
```
Comment (julio-castanar) [inline Application Lifecycle Management/instances-api.md]: Add this line with what * means. By example:
* only showed when a script executes

File: Application Lifecycle Management/instances-api.md [hunk 10]
```diff
@@ -1740,189 +2289,601 @@ ElasticBox-Release: 4.0
...
-
 Content-Type: application/json
 Authorization: Bearer your_json_web_token
 ElasticBox-Release: 4.0
 ```
 
-```
-Body:
+#### URI Parameters
+* None
```
Comment (julio-castanar) [inline Application Lifecycle Management/instances-api.md]: #### URI Parameters
Parameter |Type |Description | Req. |
|-----------|------|------------| ---- |
| instance_id | string | Instance id | Yes |

File: Application Lifecycle Management/instances-api.md [hunk 11]
```diff
@@ -1933,33 +2894,54 @@ Body:
...
```
Comment (julio-castanar) [inline Application Lifecycle Management/instances-api.md]: #### URI Parameters
Parameter |Type |Description | Req. |
|-----------|------|------------| ---- |
| instance_id | string | Instance id | Yes |

**Comment by jim-phillips** [comment]
@julio-castanar or @gsanchezu please let me know when this is approved.

File: Application Lifecycle Management/instances-api.md [hunk 12]
```diff
@@ -1740,226 +2211,644 @@ ElasticBox-Release: 4.0
...
 ]
 ```
 
-### PUT /services/instances/{instance_id}/deploy
 
-Re-deploy an existing instance, requires the specified id instance_id.
+## PUT /services/instances/{instance_id}/deploy
 
-**Normal Response Codes**
+Re-deploy an existing instance, requires the specified id instance_id.
```
Comment (julio-castanar) [inline Application Lifecycle Management/instances-api.md]: Re-deploys an existing instance. It requires the specified id instance_id.

File: Application Lifecycle Management/instances-api.md [hunk 13]
```diff
@@ -1740,226 +2211,644 @@ ElasticBox-Release: 4.0
...
   "id":"i-4g166v",
   "method":"reconfigure"
 }
 ```
 
-### PUT /services/instances/{instance_id}/import
+
+## PUT /services/instances/{instance_id}/import
 
 Retry to import an unregistered instance. when you give its ID.
```
Comment (julio-castanar) [inline Application Lifecycle Management/instances-api.md]: Retries to import an unregistered instance when you give its ID.

File: Application Lifecycle Management/instances-api.md [hunk 14]
```diff
@@ -1740,226 +2211,644 @@ ElasticBox-Release: 4.0
...
 {
   "id":"i-eruumb",
   "method":"import"
 }
 ```
 
-### PUT /services/instances/{instance_id}/cancel_import
+## PUT /services/instances/{instance_id}/cancel_import
 
 Cancel a failed import of an unregistered instance when you give its ID.
```
Comment (julio-castanar) [inline Application Lifecycle Management/instances-api.md]: Cancels a failed import ...

File: Application Lifecycle Management/instances-api.md [hunk 15]
```diff
@@ -1740,226 +2211,644 @@ ElasticBox-Release: 4.0
...
+
+#### Common Error Response Codes
 
+- **403** Forbidden
+- **404** Not Found
+
+
+#### Response Body
+
+```
```
Comment (julio-castanar) [inline Application Lifecycle Management/instances-api.md]: Change the last line in this document
old: * Windows: RDP into the instance to locate the log at ProgramDataElasticBoxLogselasticbox-agent.log
new: * Windows: RDP into the instance to locate the log at \ProgramData\ElasticBox\Logs\elasticbox-agent.log

**Comment by jtorresjj** [comment]
Jenkins test this please

**Comment by jtorresjj** [comment]
@gsanchezu  @julio-castanar, 

Please, review the last fixes.

File: Application Lifecycle Management/instances-api.md [hunk 16]
```diff
@@ -1183,193 +1321,227 @@ In this sample request, the instance is tagged and scheduled to terminate at a g
 | schema | string | Instance schema uri. |
 | lease | array | Schedules an instance with two parameters:<li>released. Is a true or false boolean value. False means that the operation scheduled on the instance is not executed yet.</li><li>expire. Specifies in UTC format YYYY-MM-DD HH:MM:SS.SSSSSS, the time and date for stopping an instance. It’s required only when an instance is set to terminate or shut down.</li><li>operation. Specifies an instance to stop with **shutdown** or **terminate**. When not scheduled, the instance is set to **alwayson**.</li> |
```
[the remark this answers is not in the document]
Comment (julio-castanar) [reply Application Lifecycle Management/instances-api.md]: Modify line 1299 too

File: Application Lifecycle Management/instances-api.md [hunk 17]
```diff
@@ -307,111 +325,330 @@ ElasticBox-Release: 4.0
...
+| lease | array | If scheduled, this object displays these parameters for the instance:<li>released. Is a true or false boolean value. False means that the operation scheduled on the instance is not executed yet.</li><li>expire. Specifies in UTC format YYYY-MM-DD HH:MM:SS.SSSSSS, the time and date for stopping an instance. It applies only when an instance is set to terminate or shut down.</li><li>operation. Specifies the stop operation scheduled as either **shutdown** or **terminate**. When not scheduled, the instance is set to **alwayson**.</li> |
+| bindings | array | List of instance bindings. |
+| binding | object | Binding contained in the bindings list, each binding object contains the parameters: instance and name. |
+| updated | string | Date of the last update. |
+|name | string | Instance name. |
+| service | Object | Instance service. |
+| service.type | string | Required. Can be one of these types: Linux Compute, Windows Compute and CloudFormation Service. |
```
Comment (julio-castanar) [inline Application Lifecycle Management/instances-api.md]: Please remove 'Required. ' on Response parameter service.type

File: Application Lifecycle Management/instances-api.md [hunk 18]
```diff
@@ -1140,7 +1287,7 @@ In this sample request, the instance is tagged and scheduled to terminate at a g
 | workflow.event | string |	Workflow action event.|
 | workflow.script | string | Workflow action script uri. |
 | tags | array | Instance tags. |
```
Comment (julio-castanar) [inline Application Lifecycle Management/instances-api.md]: Please remove 'Required. ' on Response parameter service.type (line 1279)

File: Application Lifecycle Management/instances-api.md [hunk 19]
```diff
@@ -747,164 +1013,187 @@ ElasticBox-Release: 4.0
...
 
-* Invalid Data (400)
-* Forbidden (403)
-* Not Found (404)
 
-**Request Parameters**
+#### Request Body Parameters
 
-| Parameter | Type |Description |
+|Parameter |Type |Description |
```
Comment (julio-castanar) [inline Application Lifecycle Management/instances-api.md]: Add in this table the Req. column too.
Then, in rows where necessary delete the 'Required' word and add yes in the new column.

**Comment by jtorresjj** [comment]
@julio-castanar 
I did review it. I have update it again. Sorry

File: Application Lifecycle Management/instances-api.md [hunk 20]
```diff
@@ -1,95 +1,113 @@
 {{{
 "title": "Instances API",
-"date": "09-01-2016",
-"author": "",
+"date": "12-28-2018",
+"author": "Jorge Torres",
 "attachments": [],
-"contentIsHTML": false
+"contentIsHTML": false,
+"keywords": ["CAM API", "instances", "ALM", "get instance", "delete instances", "binding", "deploy", "power-on", "shutdown", "terminate", "unregistere dinstance"]
```
Comment (gsanchezu) [inline Application Lifecycle Management/instances-api.md]:
```suggestion
"keywords": ["cam", "cam api", "instances", "alm", "get instance", "delete instance", "binding", "deploy", "power-on", "shutdown", "terminate", "unregistered instance"]
```

File: Application Lifecycle Management/instances-api.md [hunk 21]
```diff
@@ -747,171 +1013,194 @@ ElasticBox-Release: 4.0
...
 
-* Invalid Data (400)
-* Forbidden (403)
-* Not Found (404)
 
-**Request Parameters**
+#### Request Body Parameters
 
-| Parameter | Type |Description |
+|Parameter |Type |Description |
```
Comment (julio-castanar) [inline Application Lifecycle Management/instances-api.md]: Add in this table the Req. column too.
Then, in rows where necessary delete the 'Required' word and add yes in the new column.

File: Application Lifecycle Management/instances-api.md [hunk 22]
```diff
@@ -1,95 +1,113 @@
...
+| lease | array | Schedules an instance with three parameters: <li>released. Is a true or false boolean value. False means that the operation scheduled on the instance is not executed yet.</li><li>operation. Specifies an instance to stop with **shutdown** or **terminate**. When not scheduled, the instance is set to **alwayson**.</li><li>expire. Specifies in UTC format YYYY-MM-DD HH:MM:SS.SSSSSS, the time and date for stopping an instance. It’s required only when an instance is set to terminate or shut down.</li> | 
+| created | string | Creation date. |
 | updated |  string | Date of the last update. |
+| automatic_updates | string | One of these: major, minor, patch, off. Default off. |
 | members | array | List of members whom you shared the instance. |
 | owner | string | Instance owner. |
-| operation | string | Last operation, there are seven types of operations: deploy , shutdown , poweron , reinstall , reconfigure , terminate and terminate_service |
+| operation | string | Last operation, there are seven types of operations: deploy, shutdown , poweron , reinstall , reconfigure , terminate and terminate_service |
```
Comment (gsanchezu) [inline Application Lifecycle Management/instances-api.md]: Remove trailing spaces in all table rows

File: Application Lifecycle Management/instances-api.md [hunk 23]
```diff
@@ -1402,90 +1583,132 @@ ElasticBox-Release: 4.0
 | machine.schema | array | Machine schema uri. |
 | schema | string | Service schema uri. |
 | organization | string | The name of the organization that owns this service. |
-| deleted | string | Deleted. |
+| deleted | string | Null when the instance is available. |
+
+
```
Comment (gsanchezu) [inline Application Lifecycle Management/instances-api.md]: Remove additional blank lines

**Comment by gsanchezu** [comment]
Please @jtorresjj, push commit with latest changes for us to approve. Thanks!

**Comment by jtorresjj** [comment]
@gsanchezu  @julio-castanar , I sent the latest version with the proposed changes.

Thanks!

**Pull request merged**